zerver/models.py

```diff
@@ -896,6 +896,9 @@ def web_public_streams_enabled(self) -> bool:
             # target every open Internet service that can host files.
             return False
 
+        if not self.enable_spectator_access:
+            return False
```

Comment:
This is mainly for not allowing creation of web-public streams if `enable_spectator_access` is set to False. This also changes the message edit behavior where `access_web_public_message` raises `MissingAuthenticationError` if `enable_spectator_access` is set to False which I think is correct.

Reply:
Yes, that is correct.

Reply:
Though I think ideally we'd have a test in the `json_fetch_raw_message` suite that would have failed were `enable_spectator_access` to not have been checked.